apiserver/params/internal.go

```diff
@@ -834,3 +834,19 @@ type EntityString struct {
 type SetContainerSpecParams struct {
 	Entities []EntityString `json:"entities"`
 }
+
+// GoalStateResult holds the status of the charm's peers and
+// related units.
+type GoalStateResult struct {
+	Error   *Error `json:"error,omitempty"`
+	Unit    string `json:"unit"`
```

Comment:
We had discussed using a StringsResults struct containing the serialised YAML of the goal state.
What's in this struct doesn't make sense as it doesn't reflect the specification which says that goal state is one or more peer units' info and info for related units.
The other option is to return a more complex params struct but we can start simple.